File: src/test/java/com/example/retailstore/strategy/EmployeeDiscountStrategyTest.java
package com.example.retailstore.strategy;

import com.example.retailstore.model.Bill;
import org.junit.jupiter.api.Test;

import static com.example.retailstore.model.UserType.AFFILIATE;
import static com.example.retailstore.model.UserType.EMPLOYEE;
import static org.junit.jupiter.api.Assertions.assertEquals;

class EmployeeDiscountStrategyTest {
    @Test
    void shouldCalculateNetPayableAmountWithNoGroceries() {
        Bill bill = new Bill(1000, false, EMPLOYEE);

        EmployeeDiscountStrategy strategy = new EmployeeDiscountStrategy();

        double actual = strategy.calculateNetPayableAmount(bill);

        assertEquals(650, actual);
    }
}

File: src/test/java/com/example/retailstore/strategy/EmployeeDiscountStrategyTest.kt
package com.example.retailstore.strategy

import com.example.retailstore.model.Bill
import com.example.retailstore.model.UserType
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test


internal class EmployeeDiscountStrategyTest {
    @Test
    fun shouldCalculateNetPayableAmountWithNoGroceries() {
        val bill = Bill(1000.0, false, UserType.EMPLOYEE)
        val strategy = EmployeeDiscountStrategy()

        val actual = strategy.calculateNetPayableAmount(bill)

        assertEquals(650.0, actual)
    }
}